pkg/docker/config/config.go

```diff
@@ -0,0 +1,276 @@
+		return "", "", err
+	}
+	if username != "" && password != "" {
+		return username, password, nil
+	}
+
+	// check ~/.docker/config.json as secondary option
+	path = filepath.Join(homedir.Get(), dockerCfg, dockerCfgFileName)
+	username, password, err = findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	return username, password, nil
+}
+
+// RemoveAuthentication deletes the credentials stored in auth.json
+func RemoveAuthentication(registry string) error {
+	registry = dockerHostNameToRegistry(registry)
+	return modifyJSON(func(auths *dockerConfigFile) (bool, error) {
+		// First try cred helpers.
+		if ch, exists := auths.CredHelpers[registry]; exists {
+			if err := deleteAuthFromCredHelper(ch, registry); err != nil {
+				return false, err
+			}
+			return true, nil
+		}
+
+		if _, ok := auths.AuthConfigs[registry]; ok {
+			delete(auths.AuthConfigs, registry)
+		} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+			delete(auths.AuthConfigs, normalizeRegistry(registry))
+		} else {
+			return false, ErrNotLoggedIn
+		}
+		return true, nil
+	})
+}
+
+// dockerHostNameToRegistry converts the docker hostname to the docker registry
+func dockerHostNameToRegistry(registry string) string {
+	if registry == dockerHostname {
+		return dockerRegistry
+	}
+	return registry
+}
+
+// getPath gets the path of the auth.json file since the user can set their own path
+// by setting CRIO_AUTH_PATH
+// If CRIO_AUTH_PATH is not set, the default path is ~/.config/containers/auth.json
+func getPathToAuth() (string, error) {
+	if os.Getenv("CRIO_AUTH_PATH") != "" {
+		return filepath.Join(os.Getenv("CRIO_AUTH_PATH"), authCfgFileName), nil
+	}
+	dir := filepath.Join(homedir.Get(), authCfg)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err = os.Mkdir(dir, 0700); err != nil {
+			return "", errors.Wrapf(err, "error creating directory %q", dir)
+		}
+	}
+	return filepath.Join(dir, authCfgFileName), nil
+}
+
+// readJSONFile unmarshals the authentications stored in the auth.json file and returns it
+// or returns an empty dockerConfigFile data structure if auth.json is empty
+func readJSONFile(path string) (dockerConfigFile, error) {
+	var auths dockerConfigFile
+
+	raw, err := ioutil.ReadFile(path)
+	if os.IsNotExist(err) {
+		auths.AuthConfigs = map[string]dockerAuthConfig{}
+		return auths, nil
+	}
+
+	if err = json.Unmarshal(raw, &auths); err != nil {
+		return dockerConfigFile{}, errors.Wrapf(err, "error unmarshaling JSON at %q", path)
+	}
+
+	return auths, nil
+}
+
+// modifyJSON writes to auth.json if the dockerConfigFile has been updated
+func modifyJSON(editor func(auths *dockerConfigFile) (bool, error)) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return err
+	}
+
+	auths, err := readJSONFile(path)
+	if err != nil {
+		return errors.Wrapf(err, "error reading JSON file %q", path)
+	}
+
+	updated, err := editor(&auths)
+	if err != nil {
+		return errors.Wrapf(err, "error here")
```

Comment:
“error here”… maybe it’s not necessary to wrap the error here?

Reply:
whoops, my bad. fixed

Reply:
Not yet?